### PR TITLE
Ctor 474 use docker to test plugin 

### PR DESCRIPTION
**Pull request by Evan-Adam**
## Description
use docker to test installation of every plugins on every distrib and launch tests.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [X] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [X] 24.04.x (master)

<h2> How this pull request can be tested ? </h2>

see github actions, everything should be green

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: .github/workflows/plugins.yml [hunk 1]
```diff
@@ -243,6 +243,46 @@ jobs:
...
+
+
+      - uses: ./.github/actions/plugins-tests
+        with:
+          cache-key: ${{ github.sha }}-${{ github.run_id }}-${{ matrix.package_extension }}-${{ matrix.distrib }}
+          package-extension: ${{ matrix.package_extension }}
+          plugin-list: ${{needs.get-plugins.outputs.plugins}}
+
   deliver:
     needs: [get-environment, package]
```
Comment (kduret) [inline .github/workflows/plugins.yml:287]: `test-plugins` job should be added to the needs ?

File: .github/workflows/plugins.yml [hunk 2]
```diff
@@ -76,14 +76,14 @@ jobs:
...
-
           if [ "$PLUGINS" == '' ]; then
             echo "::notice::There are no modifications to the plugins packages"
           fi
+
         shell: bash
 
   fatpacker:
-    if: ${{ needs.get-plugins.outputs.plugins != '' }}
+    if: ${{needs.get-plugins.outputs.plugins != ''}}
```
Comment (kduret) [inline .github/workflows/plugins.yml:86]: revert this change

**Review by kduret — APPROVED** [review]

File: .github/scripts/test-all-plugins.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    try_command(cmd="chown snmp:snmp -R /var/lib/snmp/cert_indexes/", error="can't set cert_indexes folder permissions")
+
+    snmpsim_cmd = "snmpsim-command-responder --logging-method=null --agent-udpv4-endpoint=127.0.0.1:2024 --process-user=snmp --process-group=snmp --data-dir='./tests/robot' &"
+    try_command(cmd=snmpsim_cmd, error="can't launch snmp sim daemon.")
+
+
+def install_plugin(plugin, archi):
+    if archi == "deb":
+        output_status = (subprocess.run(
+                "apt install -o 'Binary::apt::APT::Keep-Downloaded-Packages=1;' -y ./" + plugin.lower() + "*.deb",
```
[the remark this answers is not in the document]
Comment (Evan-Adam) [reply .github/scripts/test-all-plugins.py:58]: I think we should keep the logs, to my knowledge GHA remove the old one to save space, and if an installation failed, we wan't to see what happened.

[the remark this answers is not in the document]
Comment (Evan-Adam) [reply .github/scripts/test-all-plugins.py]: Logs added as artifact only if the job fail.

File: .github/docker/testing/Dockerfile.testing-plugins-alma8 [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+# Install Robotframework
+dnf install -y python3.11 python3.11-pip
+pip3.11 install robotframework robotframework-examples
+# Install snmpsim
+pip3.11 install snmpsim-lextudio
+
+# Install node
+curl -fsSL https://rpm.nodesource.com/setup_21.x | bash -
+yum install -y nodejs
+# Install mockoon
```
Comment (paul-oureib) [inline .github/docker/testing/Dockerfile.testing-plugins-alma8:20]:
```suggestion

# Install mockoon
```

File: .github/docker/testing/Dockerfile.testing-plugins-alma8 [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+npm install -g -D @mockoon/cli@7.0.0
```
Comment (paul-oureib) [inline .github/docker/testing/Dockerfile.testing-plugins-alma8:21]: @sc979 is there a way to variabilize the version of the dependencies in a dockerfile via workflows and/or processes that we already have? (e.g. dependabot)

File: .github/docker/testing/Dockerfile.testing-plugins-alma8 [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+>> /etc/yum.repos.d/centreon-plugins.repo
+
+mkdir -p /var/lib/centreon/centplugins/
+chmod 777 /var/lib/centreon/centplugins/
+
+dnf clean all
+
+EOF
+
+WORKDIR /tests
```
Comment (paul-oureib) [inline .github/docker/testing/Dockerfile.testing-plugins-alma8:69]:
```suggestion
WORKDIR /tests

```

File: .github/docker/testing/Dockerfile.testing-plugins-bookworm [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+EOF
+
+ENV LANG en_US.utf8
+
+RUN bash -e <<EOF
+apt-get update
+# Install Robotframework
+apt-get install -y python3-dev python3-pip
+rm -rf /usr/lib/python3.11/EXTERNALLY-MANAGED
```
Comment (paul-oureib) [inline .github/docker/testing/Dockerfile.testing-plugins-bookworm:24]: @Evan-Adam what is this

File: .github/docker/testing/Dockerfile.testing-plugins-bookworm [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+pip3 install robotframework robotframework-examples
+# Install snmpsim
+pip3 install snmpsim-lextudio
+
+# Install nodejs
+curl -fsSL https://deb.nodesource.com/setup_21.x | bash - &&\
+apt-get install -y nodejs
+# Install mockoon (needs nodejs)
```
Comment (paul-oureib) [inline .github/docker/testing/Dockerfile.testing-plugins-bookworm:32]:
```suggestion

# Install mockoon (needs nodejs)
```

File: .github/docker/testing/Dockerfile.testing-plugins-alma9 [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+>> /etc/yum.repos.d/centreon-plugins.repo
+
+mkdir -p /var/lib/centreon/centplugins/
+chmod 777 /var/lib/centreon/centplugins/
+
+dnf clean all
+
+EOF
+
+WORKDIR /tests
```
Comment (paul-oureib) [inline .github/docker/testing/Dockerfile.testing-plugins-alma9:69]: @Evan-Adam if this is useful in this dockerfile, why not spread it in others?

File: .github/workflows/plugins.yml [hunk 10]
```diff
@@ -243,8 +243,56 @@ jobs:
...
+    container:
+      image: ${{ vars.DOCKER_INTERNAL_REGISTRY_URL }}/${{ matrix.image }}
+      credentials:
+        username: ${{ secrets.DOCKER_REGISTRY_ID }}
+        password: ${{ secrets.DOCKER_REGISTRY_PASSWD }}
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+
```
Comment (paul-oureib) [inline .github/workflows/plugins.yml:279]:
```suggestion
```

File: .github/workflows/plugins.yml [hunk 11]
```diff
@@ -243,8 +243,56 @@ jobs:
...
+      - uses: ./.github/actions/test-plugins
+        with:
+          cache-key: ${{ github.sha }}-${{ github.run_id }}-${{ matrix.package_extension }}-${{ matrix.distrib }}
+          package-extension: ${{ matrix.package_extension }}
+          plugin-list: ${{needs.get-plugins.outputs.plugins}}
```
Comment (paul-oureib) [inline .github/workflows/plugins.yml:284]:
```suggestion
          plugin-list: ${{ needs.get-plugins.outputs.plugins }}
```

File: .github/workflows/plugins.yml [hunk 12]
```diff
@@ -243,8 +243,56 @@ jobs:
...
+
+      - name: Upload apt/dnf logs as artifacts if tests failed.
```
Comment (paul-oureib) [inline .github/workflows/plugins.yml:286]:
```suggestion
      - name: Upload apt/dnf logs as artifacts if tests failed
```

File: .github/docker/testing/Dockerfile.testing-plugins-alma9 (hunk 9 above)
Comment (Evan-Adam) [inline .github/docker/testing/Dockerfile.testing-plugins-alma9:69]:
```suggestion
```

File: .github/docker/testing/Dockerfile.testing-plugins-bookworm (hunk 7 above)
Comment (Evan-Adam) [reply .github/docker/testing/Dockerfile.testing-plugins-bookworm:24]: needed to install library with pip, without it we have this error 
`
63.32 error: externally-managed-environment
63.32
63.32 × This environment is externally managed
63.32 ╰─> To install Python packages system-wide, try apt install
63.32     python3-xyz, where xyz is the package you are trying to
63.32     install.
63.32
63.32     If you wish to install a non-Debian-packaged Python package,
63.32     create a virtual environment using python3 -m venv path/to/venv.
63.32     Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
63.32     sure you have python3-full installed.
63.32
63.32     If you wish to install a non-Debian packaged Python application,
63.32     it may be easiest to use pipx install xyz, which will manage a
63.32     virtual environment for you. Make sure you have pipx installed.
63.32
63.32     See /usr/share/doc/python3.11/README.venv for more information.
63.32
63.32 note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
63.32 hint: See PEP 668 for the detailed specification.
`

File: .github/docker/testing/Dockerfile.testing-plugins-alma9 (hunk 9 above)
Comment (Evan-Adam) [reply .github/docker/testing/Dockerfile.testing-plugins-alma9]: removed.

File: .github/docker/testing/Dockerfile.testing-plugins-alma8 (hunk 5 above)
Comment (Evan-Adam) [reply .github/docker/testing/Dockerfile.testing-plugins-alma8:22]: CTOR-564 created to address this problem.

Comment (sc979) [reply .github/docker/testing/Dockerfile.testing-plugins-alma8]: Everything is always possible. But why should we variabilize this testing image's dependency version ?

Comment (Evan-Adam) [reply .github/docker/testing/Dockerfile.testing-plugins-alma8]: tested the last version of mockoon, I didn't see any bug.
I let the last mockoon version in the build of images so it will be updated automaticly on image rebuild.

**Review by paul-oureib — APPROVED** [review]
